### PR TITLE
Pmd7 upgrade 7.2.0 sql set 1

### PR DESCRIPTION
**Pull request by jborgers**
First two sql rules.
Created #375 for second.

### CODE REVIEW

File: src/main/resources/category/java/sql.xml [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+]
+]]>
+                </value>
+            </property>
+        </properties>
+        <example>
+            <![CDATA[
+    public SomeEntity findByQueryBad(TypedQuery<Object> query) {
+        final List<SomeEntity> entities = query.getResultList();
+        return !entities.isEmpty() ? entities.get(0) : null;
```
Comment (stokpop) [inline src/main/resources/category/java/sql.xml:156]: Note there are other ways to get first element: 

- java 21 has `getFirst()`, which intellij also suggests as improved code. 
- via iterator: `entities.iterator().next()`
- via stream: `entities.stream().findFirst().get()` or other stream constructs with `findAny()`

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml [hunk 2]
```diff
@@ -68,29 +74,17 @@ public class Foo {
         javax.persistence.Query q = entityManager.createQuery("SELECT reas FROM Information info WHERE reas.code = :code", Object.class);
         q.setParameter("code", code);
         q.setMaxResults(1); //added JB
-        q.setHint("org.hibernate.fetchSize", "1000"); // bad: added JB option 1 - should violate
-        q.setHint("org.hibernate.fetchSize", FETCH_SIZE_HUGE); // bad: added JB option 1 - should violate
+        q.setHint("org.hibernate.fetchSize", "1000"); // bad: added JB option 1 - should violate @TODO
```
Comment (stokpop) [inline src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml:77]: why `@TODO`? (2x)

**Review by stokpop — CHANGES_REQUESTED** [review]
Two remarks/questions. TODO's can be removed or explained?

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml (hunk 2 above)
Comment (stokpop) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml:77]: Ah, is this the #375 issue?

File: src/main/resources/category/java/sql.xml (hunk 1 above)
Comment (stokpop) [reply src/main/resources/category/java/sql.xml:156]: Maybe getFirst() match can be added ... others via issue for future?

File: src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml (hunk 2 above)
Comment (jborgers) [reply src/test/resources/com/jpinpoint/perf/lang/java/ruleset/sql/xml/AvoidHugeQueryFetchSize.xml:77]: Indeed, I will add it

File: src/main/resources/category/java/sql.xml (hunk 1 above)
Comment (jborgers) [reply src/main/resources/category/java/sql.xml:156]: okay, good to know, new functionality, for improvement request

Comment (jborgers) [reply src/main/resources/category/java/sql.xml:156]: Added #377

**Pull request merged**